haystack/document_stores/pinecone.py

```diff
@@ -0,0 +1,721 @@
+        index = self._sanitize_index_name(index)
+        documents = super(PineconeDocumentStore, self).get_all_documents_generator(
+            index=index, filters=filters, batch_size=batch_size, return_embedding=False
+        )
+
+        for doc in documents:
+            if return_embedding:
+                self._attach_embedding_to_document(document=doc, index=index)
+            yield doc
+
+    def get_documents_by_id(
+        self,
+        ids: List[str],
+        index: Optional[str] = None,
+        batch_size: int = 32,
+        headers: Optional[Dict[str, str]] = None,
+        return_embedding: Optional[bool] = None,
+    ) -> List[Document]:
+
+        if headers:
+            raise NotImplementedError("PineconeDocumentStore does not support headers.")
+
+        if return_embedding is None:
+            return_embedding = self.return_embedding
+
+        index = index or self.index
+        index = self._sanitize_index_name(index)
+
+        documents = super().get_documents_by_id(ids=ids, index=index, batch_size=batch_size)
+        if return_embedding:
+            for doc in documents:
+                self._attach_embedding_to_document(document=doc, index=index)
+
+        return documents
+
+    def get_embedding_count(
+        self, index: Optional[str] = None, filters: Optional[Dict[str, Union[Dict, List, str, int, float, bool]]] = None
+    ) -> int:
+        """
+        Return the count of embeddings in the document store.
+        """
+        if filters:
+            raise NotImplementedError("Filters are not supported for get_embedding_count in PineconeDocumentStore")
+
+        index = index or self.index
+        index = self._sanitize_index_name(index)
+        if not self.pinecone_indexes.get(index, False):
+            raise ValueError(f"No index named {index} found in Pinecone.")
+
+        stats = self.pinecone_indexes[index].describe_index_stats()
+        # if no namespace return zero
+        count = stats["namespaces"][""]["vector_count"] if "" in stats["namespaces"] else 0
+        return count
+
+    def update_document_meta(self, id: str, meta: Dict[str, str], index: str = None):
+        """
+        Update the metadata dictionary of a document by specifying its string id
+        """
+        index = index or self.index
+        index = self._sanitize_index_name(index)
+        if index in self.pinecone_indexes:
+            doc = self.get_documents_by_id(ids=[id], index=index, return_embedding=True)[0]
+            if doc.embedding is not None:
+                self.pinecone_indexes[index].upsert(vectors=([id], [doc.embedding.tolist()], [meta]))
+
+        super().update_document_meta(id=id, meta=meta, index=index)
+
+    def delete_documents(
+        self,
+        index: Optional[str] = None,
+        ids: Optional[List[str]] = None,
+        filters: Optional[Dict[str, Union[Dict, List, str, int, float, bool]]] = None,
+        headers: Optional[Dict[str, str]] = None,
```

Comment:
Similar as before, I'd say if the method has a parameter `headers` we should describe it with `:param headers: Not implemented...` otherwise a user only finds out when seeing the Exception raised.